### PR TITLE
Document debug logging.

### PR DESCRIPTION
**Pull request by swankjesse**
Also refine the formatting of TaskRunner logging.

Before:

    [2019-12-30 16:43:19] Q1 starting: OkHttp ConnectionPool
    [2019-12-30 16:43:19] Q1 scheduled after 300 s: OkHttp ConnectionPool
    [2019-12-30 16:43:19] Q1 finished in 94 µs: OkHttp ConnectionPool
    [2019-12-30 16:43:19] Q359 scheduled after 0 µs: OkHttp www.youtube.com applyAndAckSettings
    [2019-12-30 16:43:19] Q359 starting: OkHttp www.youtube.com applyAndAckSettings
    [2019-12-30 16:43:19] Q361 schedule canceled (queue is shutdown): OkHttp www.youtube.com onSettings
    [2019-12-30 16:43:19] Q359 finished in 382 µs: OkHttp www.youtube.com applyAndAckSettings
    [2019-12-30 16:43:20] Q1 scheduled after 0 µs: OkHttp ConnectionPool
    [2019-12-30 16:43:20] Q365 scheduled after 0 µs: OkHttp android-developers.googleblog.com applyAndAckSettings
    [2019-12-30 16:43:20] Q1 starting: OkHttp ConnectionPool
    [2019-12-30 16:43:20] Q365 starting: OkHttp android-developers.googleblog.com applyAndAckSettings
    [2019-12-30 16:43:20] Q1 scheduled after 300 s: OkHttp ConnectionPool
    [2019-12-30 16:43:20] Q1 finished in 151 µs: OkHttp ConnectionPool

After:

    [2019-12-30 16:43:19] Q10001 starting              : OkHttp ConnectionPool
    [2019-12-30 16:43:19] Q10001 run again after 300 s : OkHttp ConnectionPool
    [2019-12-30 16:43:19] Q10001 finished run in  94 µs: OkHttp ConnectionPool
    [2019-12-30 16:43:19] Q10359 scheduled after   0 µs: OkHttp www.youtube.com applyAndAckSettings
    [2019-12-30 16:43:19] Q10359 starting              : OkHttp www.youtube.com applyAndAckSettings
    [2019-12-30 16:43:19] Q10361 schedule canceled (queue is shutdown): OkHttp www.youtube.com onSettings
    [2019-12-30 16:43:19] Q10359 finished run in 382 µs: OkHttp www.youtube.com applyAndAckSettings
    [2019-12-30 16:43:20] Q10001 scheduled after   0 µs: OkHttp ConnectionPool
    [2019-12-30 16:43:20] Q10365 scheduled after   0 µs: OkHttp android-developers.googleblog.com applyAndAckSettings
    [2019-12-30 16:43:20] Q10001 starting              : OkHttp ConnectionPool
    [2019-12-30 16:43:20] Q10365 starting              : OkHttp android-developers.googleblog.com applyAndAckSettings
    [2019-12-30 16:43:20] Q10001 run again after 300 s : OkHttp ConnectionPool
    [2019-12-30 16:43:20] Q10001 finished run in 151 µs: OkHttp ConnectionPool

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/concurrent/TaskLogger.kt [hunk 1]
```diff
@@ -48,30 +48,34 @@ internal inline fun <T> logElapsed(
     if (loggingEnabled) {
       val elapsedNs = queue.taskRunner.backend.nanoTime() - startNs
       if (completedNormally) {
-        log(task, queue, "finished in ${formatDuration(elapsedNs)}")
+        log(task, queue, "finished run in ${formatDuration(elapsedNs)}")
       } else {
-        log(task, queue, "failed in ${formatDuration(elapsedNs)}")
+        log(task, queue, "failed a run in ${formatDuration(elapsedNs)}")
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/concurrent/TaskLogger.kt]: carefully chosen words to get the right width!

File: docs/debug_logging.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+Debug Logging
+=============
+
+OkHttp has internal APIs to enable debug logging. It uses the `java.util.logging` API which can be
+tricky to configure. As a shortcut, you can paste [OkHttpDebugLogging.kt]. Then enable debug logging
+for whichever features you need:
+
```
Comment (yschimke) [inline docs/debug_logging.md]: Probably worth a sentence on Android, example logcat filter etc?

**Comment by yschimke** [comment]
Nice user friendly change!

File: docs/debug_logging.md (hunk 2 above)
Comment (swankjesse) [reply docs/debug_logging.md]: Added. Turns out hooking up logcat to java.util.Logging nicely is a bit of a pain if you want to use `FINE`!

Comment (swankjesse) [reply docs/debug_logging.md]: (Good idea though!)

**Pull request merged**